tests/integration/test_parsers.py

```diff
@@ -0,0 +1,113 @@
+from circuit_maintenance_parser.parsers.lumen import ParserLumen
+from circuit_maintenance_parser.parsers.ntt import ParserNTT
+from circuit_maintenance_parser.parsers.packetfabric import ParserPacketFabric
+from circuit_maintenance_parser.parsers.telstra import ParserTelstra
+from circuit_maintenance_parser.parsers.zayo import ParserZayo
+
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+
+@pytest.mark.parametrize(
+    "parser, raw_file, results_file",
+    [
+        # EUNetworks
+        (
+            ParserEUNetworks,
+            Path(dir_path, "data", "eunetworks", "eunetworks1"),
+            Path(dir_path, "data", "eunetworks", "eunetworks1_result.json"),
+        ),
+        # iCal
+        (ICal, Path(dir_path, "data", "ical", "ical1"), Path(dir_path, "data", "ical", "ical1_result.json"),),
+        (ICal, Path(dir_path, "data", "ical", "ical2"), Path(dir_path, "data", "ical", "ical2_result.json"),),
+        (ICal, Path(dir_path, "data", "ical", "ical3"), Path(dir_path, "data", "ical", "ical3_result.json"),),
+        (ICal, Path(dir_path, "data", "ical", "ical4"), Path(dir_path, "data", "ical", "ical4_result.json"),),
+        (ICal, Path(dir_path, "data", "ical", "ical5"), Path(dir_path, "data", "ical", "ical5_result.json"),),
+        # Lumen
+        (
+            ParserLumen,
+            Path(dir_path, "data", "lumen", "lumen1.html"),
+            Path(dir_path, "data", "lumen", "lumen1_result.json"),
+        ),
+        (
+            ParserLumen,
+            Path(dir_path, "data", "lumen", "lumen2.html"),
+            Path(dir_path, "data", "lumen", "lumen2_result.json"),
+        ),
+        # NTT
+        (ParserNTT, Path(dir_path, "data", "ntt", "ntt1"), Path(dir_path, "data", "ntt", "ntt1_result.json"),),
+        # PacketFabric
+        (
+            ParserPacketFabric,
+            Path(dir_path, "data", "packetfabric", "packetfabric1"),
+            Path(dir_path, "data", "packetfabric", "packetfabric1_result.json"),
+        ),
+        # Telstra
+        (
+            ParserTelstra,
+            Path(dir_path, "data", "telstra", "telstra1.html"),
+            Path(dir_path, "data", "telstra", "telstra1_result.json"),
+        ),
+        (
+            ParserTelstra,
+            Path(dir_path, "data", "telstra", "telstra2.html"),
+            Path(dir_path, "data", "telstra", "telstra2_result.json"),
+        ),
+        # Zayo
+        (
+            ParserZayo,
+            Path(dir_path, "data", "zayo", "zayo1.html"),
+            Path(dir_path, "data", "zayo", "zayo1_result.json"),
+        ),
+        (
+            ParserZayo,
+            Path(dir_path, "data", "zayo", "zayo2.html"),
+            Path(dir_path, "data", "zayo", "zayo2_result.json"),
+        ),
+    ],
+)
+def test_complete_parsing(parser, raw_file, results_file):
+    """Tests parser."""
+    with open(raw_file, "rb") as file_obj:
+        parser = parser(raw=file_obj.read())
```

Comment:
I don't like redefining the `parser` variable like this. Maybe rename the function parameter to `parser_class` and reserve `parser` for the instance you're creating here? Same comment on the other test function below.

Reply:
absolutely, I did a quick replace and I missed that point